litmus-portal/backend/self-deployer/pkg/util/uitls.go

```diff
@@ -0,0 +1,101 @@
+package util
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/litmuschaos/litmus/litmus-portal/backend/self-deployer/pkg/k8s"
+
+)
+
+func WaitForServer(server string) {
+	log.Print("Checking if Server Up ", server)
+	for true {
+		req, err := http.NewRequest("GET", server, bytes.NewBuffer([]byte("")))
+		req.Header.Set("Content-Type", "application/json")
+		client := &http.Client{}
+		res, err := client.Do(req)
+		if err != nil || res.StatusCode != 200 {
+			time.Sleep(30 * time.Second)
+		} else {
+			break
+		}
+	}
+}
+
+func RegSelfCluster(server, pid string) (string, error) {
+	var jsonStr = []byte(`{"query":"mutation { userClusterReg(clusterInput: { cluster_name: \"Self-Cluster\", description:\"Self-Cluster\", project_id:\"` + pid + `\", platform_name:\"others\", cluster_type:\"INTERNAL\"})}"}`)
```

Comment:
Put the graphql query in a string var and remove the escape character to look like a normal string. So that it will be easy to understand the query.

Reply:
The escape characters is necessary, it throws errors without those

Reply:
You have to convert string to escape string , before passing to the function